client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -89,6 +95,11 @@ export function SitesDashboard( { queryParams: { search, status } }: SitesDashbo
 
 	const selectedTabName = tabs.find( ( tab ) => tab.name === status )?.name;
 
+	const { sortedSites } = useSitesTableSorting( filteredSites, {
```

Comment:
This is running the sorting after the filtering, but I think it could actually be more efficient to do the sorting first. That way we will only resort when the data from the query changes rather than every time the user changes the filtering.

Reply:
@p-jackson as far as I see, the whole logic (useSiteExcerptsQuery, useSitesTableFiltering) is executed on each tab change and on each search query change. If we do sorting first, wouldn't we sort the whole dataset each time?

Reply:
If sorting was done first then on each tab change:
- `useSiteExcerptsQuery` is called, but the underlying `useQuery` hook will fetch data from cache so this is quick
- then `useSitesTableSorting` is called, but since the result from `useSiteExcerptsQuery` hasn't changed `useMemo`  will return the pre-sorted data from memory
- then `useSitesTableFiltering` is called, but since `useSitesTableSorting` is returning an identical set of sites it doesn't retrigger filtering and the `useMemo` that does the filtering will return the exact same pre-filtered set of sites.

(this is assuming `useSitesTableSorting` is using `useMemo` under the hood)

Reply:
I see, I adjusted that. I needed to disable fuzzy search sorting too as it was sorting records by relevance.

Reply:
Hmmm I think we might need that fuzzy search sorting to come back. It's still good to run the `useSitesTableSorting` hook first because it saves on sorting when something else (other than searching) has to re-render. But I find the fuzzy search is quite counterintuitive without sorting by relevance. Sometimes it means that a site that has very little to do with my search term appears at the top of the list, even when there's a perfect match available in the table.